src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -756,6 +758,9 @@ void MutationsInterpreter::prepare(bool dry_run)
 
                 if (condition && settings.return_mutated_rows)
                     stages.back().filters.push_back(condition);
+
+                if (type == RowExistsColumn::type)
+                    update_where_condition = condition;
```

Comment:
It is incorrect because one mutation can contain several commands with different predicates and even different commands, and some may not be supported for projections. You need to collect identifiers from all predicates and do not allow executing mutations in the presence of unsupproted commands.

Reply:
Also the name `update_where_condition` is bad because it doesn't reflect lightweight deletes anyhow but is used only for lightweight deletes.

Reply:
However, you don't need this info in `MutationsInterpreter` at all, since you can get it from MutationCommands. I suggest adding corresponding methods into `MutationCommands`

Reply:
So I guess to be conservative, I can check the condition that only command of update and row_exists column are involved at the same time.  But could you give an example that a mutation involves several commands so I can test against it?